### PR TITLE
feat: implement standardize() for spectral data normalization

### PR DESCRIPTION
**Pull request by S-Leuthold**
## Summary
- Implements `standardize()` function for spectral data normalization
- Handles resampling, trimming, water band removal, and baseline correction
- Operations applied in optimal order: trim → baseline → resample → water bands

## Key Features
- **Resampling**: Spline interpolation via prospectr to target resolution (default 2 cm⁻¹ for OSSL)
- **Trimming**: Subset to wavenumber range with validation
- **Water band removal**: OSSL standard ranges (1580-1720, 3100-3700 cm⁻¹)
- **Baseline correction**: Convex hull via prospectr with error handling
- **Idempotent**: Warns if already standardized, supports force re-standardization
- **Provenance tracking**: Records all parameters and results

## Test Coverage
- 48 tests covering input validation, all operations, edge cases
- Tests for error conditions (trim out of range, water bands delete all, etc.)

## Files Changed
- `R/pipeline-standardize.R` (new)
- `tests/testthat/test-pipeline-standardize.R` (new)

### CODE REVIEW

File: man/standardize.Rd [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+\item{x}{\link{horizons_data.} Object from \code{spectra()}.}
+
+\item{resample}{\link{numeric or NULL.} Target resolution in cm⁻¹. Default \code{2}
+matches OSSL library resolution. Use \code{NULL} to skip resampling.}
+
+\item{trim}{\link{numeric(2) or NULL.} Wavenumber range to keep, e.g.,
+\code{c(600, 4000)}. Default \code{NULL} skips trimming.}
+
+\item{remove_water}{\link{logical.} Remove water absorption bands
+(~1600-1700, ~3200-3600 cm⁻¹)? Default \code{FALSE}.}
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply man/standardize.Rd]: ✅ Fixed — updated roxygen to match WATER_BANDS constant (1580-1720, 3100-3700 cm⁻¹).

File: R/class-core.R [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
+  if (length(numeric_predictors) > 1) {
+
+    wn_values <- as.numeric(numeric_predictors)
+
+    if (!all(diff(wn_values) < 0)) {
+
+      errors <- c(errors, cli::format_inline("Wavelength columns must be in decreasing order"))
+
+    }
+  }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/class-core.R:359]: Dismissing — this is intentional by design. Bare numeric columns come from raw CSV input; wn_-prefixed columns are pipeline-produced and guaranteed ordered. The regex distinction is the design, not a bug.

File: R/class-core.R [hunk 3]
```diff
@@ -0,0 +1,815 @@
...
+        branch      <- if (has_covars) "\u251C\u2500" else "\u2514\u2500"
+        cat(paste0("   ", branch, " Predictors: ", x$data$n_predictors, range_str, "\n"))
+
+      } else {
+
+        has_covars <- !is.null(x$data$n_covariates) && x$data$n_covariates > 0
+        branch     <- if (has_covars) "\u251C\u2500" else "\u2514\u2500"
+        cat(paste0("   ", branch, " Predictors: ", x$data$n_predictors, "\n"))
+
+      }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/class-core.R:504]: ✅ Fixed — regex now matches both bare numeric and wn_-prefixed columns via `^(wn_)?[0-9.]+$`, with gsub to strip prefix before computing range.

File: R/class-core.R [hunk 4]
```diff
@@ -0,0 +1,815 @@
...
+      if (length(numeric_wn) > 1) {
+
+        wn_values <- sort(as.numeric(numeric_wn), decreasing = TRUE)
+        wn_range  <- range(wn_values)
+        wn_step   <- abs(diff(wn_values[1:2]))
+
+        cat(paste0("   \u2502     \u251C\u2500 Range: ", wn_range[2], "\u2013", wn_range[1], " cm\u207B\u00B9\n"))
+        cat(paste0("   \u2502     \u2514\u2500 Step: ", wn_step, " cm\u207B\u00B9\n"))
+
+      }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/class-core.R:637]: ✅ Fixed — same regex fix applied to summary() method.

File: R/class-core.R [hunk 5]
```diff
@@ -0,0 +1,815 @@
...
+
+      predictor_vars <- x$data$role_map$variable[x$data$role_map$role == "predictor"]
+      numeric_wn     <- predictor_vars[grepl("^[0-9.]+$", predictor_vars)]
+
+      has_covariates <- !is.null(x$data$n_covariates) && x$data$n_covariates > 0
+      has_outcome    <- any(x$data$role_map$role == "outcome")
+      has_more       <- has_covariates || has_outcome
+      branch         <- if (has_more) "\u251C\u2500" else "\u251C\u2500"
+
+      cat(paste0("   ", branch, " Predictors: ", x$data$n_predictors, "\n"))
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/class-core.R:626]: ✅ Fixed — branch now uses └─ when no more items follow (was ├─ in both arms).

File: R/class-core.R [hunk 6]
```diff
@@ -0,0 +1,815 @@
...
+    ## Covariates with names ----
+
+    if (!is.null(x$data$n_covariates) && x$data$n_covariates > 0) {
+
+      covariate_vars <- x$data$role_map$variable[x$data$role_map$role == "covariate"]
+      covar_list     <- paste(covariate_vars, collapse = ", ")
+      has_outcome    <- any(x$data$role_map$role == "outcome")
+      branch         <- if (has_outcome) "\u251C\u2500" else "\u251C\u2500"
+
+      cat(paste0("   ", branch, " Covariates: ", x$data$n_covariates, "\n"))
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/class-core.R:649]: ✅ Fixed — same branch fix applied to covariates section in summary().

File: R/pipeline-spectra.R [hunk 7]
```diff
@@ -0,0 +1,1205 @@
...
+        if (!is.matrix(spectral_data) || nrow(spectral_data) != 1) {
+
+          cli::cli_warn(c(
+            "Unexpected data structure in {.file {basename(file_path)}}",
+            "i" = "Channel '{channel}' has {nrow(spectral_data)} rows, expected 1",
+            "i" = "Skipping this file"
+          ))
+          next
+
+        }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/pipeline-spectra.R:444]: Dismissing — `next` is inside the inner `for (channel in channel_priority)` loop, not the outer file loop. It correctly skips to the next channel candidate when data structure is unexpected. The outer loop continues to the next file via `found_channel` check. No bug here.

File: R/pipeline-standardize.R [hunk 8]
```diff
@@ -0,0 +1,749 @@
...
+      operations$resample <- list(
+        resolution_before = current_resolution,
+        resolution_after  = resample,
+        n_before          = result$n_before,
+        n_after           = result$n_after
+      )
+
+    }
+
+  }
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/pipeline-standardize.R:635]: Deferred — nice-to-have but not blocking. Tracked for a follow-up pass.

File: R/pipeline-standardize.R [hunk 9]
```diff
@@ -0,0 +1,749 @@
...
+
+  ## ---------------------------------------------------------------------------
+  ## Step 8: Rebuild role_map
+  ## ---------------------------------------------------------------------------
+
+  new_role_map <- tibble::tibble(
+    variable = c(non_predictor_vars, new_predictor_names),
+    role     = c(role_map$role[role_map$role != "predictor"],
+                 rep("predictor", length(new_predictor_names)))
+  )
```
[the remark this answers is not in the document]
Comment (S-Leuthold) [reply R/pipeline-standardize.R:697]: Deferred — good catch on canonical column order. Tracked for a follow-up pass.
